services/vehicle/src/main/java/com/wedriveu/services/vehicle/boundary/nearest/NearestConsumerVerticle.java

```diff
@@ -0,0 +1,62 @@
+package com.wedriveu.services.vehicle.boundary.nearest;
+
+import com.wedriveu.services.shared.rabbitmq.VerticleConsumer;
+import com.wedriveu.services.shared.utilities.Constants;
+import com.wedriveu.services.vehicle.rabbitmq.Messages;
+import io.vertx.core.json.JsonObject;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Starts the consumer which handles the user request for the nearest vehicle.
+ * The information the consumer expects to receive are the user current position (Latitude, Longitude), the chosen
+ * destination position (Latitude, Longitude) and the username associated with the user.
+ *
+ * @author Marco Baldassarri
+ * @since 30/07/2017
+ */
+public class NearestConsumerVerticle extends VerticleConsumer {
+
+    public NearestConsumerVerticle() {
+        super(Constants.CONSUMER_VEHICLE_SERVICE);
+    }
+
+    @Override
+    public void start() throws Exception {
+        super.start();
+        startUserConsumer();
```

Comment:
Don't you need to wait until this operation completes?
If so, you have to override the `start(Future<Void> startFuture)` of the `AbstractVerticle` class, and maybe compose the `startUserConsumer` operations in a sequential way, see [http://vertx.io/docs/vertx-core/java/#_sequential_composition](http://vertx.io/docs/vertx-core/java/#_sequential_composition)